botorch/optim/parameter_constraints.py

```diff
@@ -312,13 +312,44 @@ def _make_linear_constraints(
     return constraints
 
 
+def _generate_unfixed_nonlin_constraints(
+    constraints: Optional[List[Callable]],
+    fixed_features: Dict[int, float],
+    dimension: int,
+) -> Optional[List[Callable]]:
+    # If constraints is None or an empty list, then return itself
+    if not constraints:
+        return constraints
+
+    selector = []
+    idx_X, idx_f = 0, dimension - len(fixed_features)
+    for i in range(dimension):
+        if i in fixed_features.keys():
+            selector.append(idx_f)
+            idx_f += 1
+        else:
+            selector.append(idx_X)
+            idx_X += 1
+
+    values = torch.tensor(list(fixed_features.values()), dtype=torch.double)
+
+    def _wrap_nlc(nlc: Callable) -> Callable:
```

Comment:
Complete type signature and more explicit naming.
```suggestion
    def _wrap_nonlin_constraint(constraint: Callable[[Tensor], Tensor]) -> Callable[[Tensor], Tensor]:
```